File: problem_solving/easy/compare_the_triplets/compare_with_triplets.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the compareTriplets function below.
def compareTriplets(a, b)
res = [0, 0]
    a.each_with_index do |e, i| 
        if e > b[i]
            res[0] += 1
        elsif e < b[i] 
            res[1] += 1
        end
    end  
    res    
end

#fptr = File.open(ENV['OUTPUT_PATH'], 'w')
fptr = File.open(result.txt, 'w')

a = gets.rstrip.split.map(&:to_i)

b = gets.rstrip.split.map(&:to_i)

result = compareTriplets a, b

fptr.write result.join " "
fptr.write "\n"

fptr.close()

File: problem_solving/easy/compare_the_triplets/compare_with_triplets.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

const arrSq = (a, b) => a.map((elem, i) => [a[i], b[i]]);
const arrCmp = ([sa, sb], [a, b]) => (
    [
        sa += a > b ? 1 : 0,
        sb += b > a ? 1 : 0
    ]
);

// Complete the compareTriplets function below.
function compareTriplets(a, b) {
    return arrSq(a, b).reduce(arrCmp, [0, 0]);
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const a = readLine().replace(/\s+$/g, '').split(' ').map(aTemp => parseInt(aTemp, 10));

    const b = readLine().replace(/\s+$/g, '').split(' ').map(bTemp => parseInt(bTemp, 10));

    const result = compareTriplets(a, b);

    ws.write(result.join(' ') + '\n');

    ws.end();
}
